### PR TITLE
[v1.x] ONNX support adaptiveAveragePooling2D and update Softmax to support temperature

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Support onnx conversion of `_contrib_AdaptiveAvgPooling2D`. Update onnx conversion of `softmax` to support `temperature` attribute.

Due to some incompatible issue of mx-onnx conversion, currently we only support the conversion when `output_size==1`. Pytorch has the similar implementation: https://github.com/pytorch/pytorch/pull/9711.

By inspecting all models in gluon model zoo, 16 of them has `_contrib_AdaptiveAvgPooling2D`. Out of the 16 models, 7 models use `output_size>1`, which are not supported in this pr. 
Unsupported models:
`psp_resnet50_ade`
`psp_resnet101_ade`
`psp_resnet101_voc`
`psp_resnet101_citys`
`psp_resnet101_coco`
`icnet_resnet50_mhpv1`
`icnet_resnet50_citys`

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -2725,3 +2725,22 @@ def convert_arange(node, **kwargs):
...
+
+    output_size = attrs.get('output_size', '1')
+    output_size = convert_string_to_list(output_size)
+
+    if len(output_size) <= 2:
+        if output_size[0] != 1 or (len(output_size) == 2 and output_size[1] != 1):
+            raise NotImplementedError("_contrib_AdaptiveAvgPooling2D operator with output_size != 1 \
+                                not yet implemented.")
+    nodes = [
+        make_node("GlobalAveragePool", [input_nodes[0]], [name])
```
Comment (josephevans) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:2744]: I think you also need to assign name=name here as the last node.

Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:2744]: Good point, thanks!

**Comment by waytrue17** [comment]
@mxnet-bot run ci [unix-cpu]

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 2]
```diff
@@ -3023,3 +3031,22 @@ def convert_contrib_box_decode(node, **kwargs):
...
+def convert_contrib_AdaptiveAvgPooling2D(node, **kwargs):
+    """Map MXNet's _contrib_BilinearResize2D operator attributes to onnx's operator.
+    """
+    from onnx.helper import make_node
+    name, input_nodes, attrs = get_inputs(node, kwargs)
+
+    output_size = attrs.get('output_size', '1')
+    output_size = convert_string_to_list(output_size)
+
+    if len(output_size) <= 2:
```
Comment (Zha0q1) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3045]: Why do we need to check this?

Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3045]: Yes, when `len(output_size) > 2`, mxnet will perform the calculation with ` output_size = 1`. This check is to handle that.

File: tests/python-pytest/onnx/test_operators.py [hunk 3]
```diff
@@ -338,17 +338,18 @@ def test_onnx_export_cast(tmp_path, src_dtype, dst_dtype, shape):
 
 
 @pytest.mark.parametrize('dtype', ['float16', 'float32'])
-def test_onnx_export_softmax(tmp_path, dtype):
+@pytest.mark.parametrize('temperature', [.1, 1., 10.])
```
Comment (josephevans) [inline tests/python-pytest/onnx/test_operators.py:341]: Should we check when temperature is not provided (and should default to 1.0?)

**Review by josephevans — APPROVED** [review]

File: tests/python-pytest/onnx/test_operators.py (hunk 3 above)
Comment (waytrue17) [reply tests/python-pytest/onnx/test_operators.py:341]: Removed temperature from M1 to test the default case.
